python/lsst/ts/electrometer/commander.py

```diff
@@ -0,0 +1,72 @@
+import asyncio
+import logging
+from lsst.ts import tcpip
+
+
+class TCPIPCommander:
+    """Implement communication with the electrometer.
+
+    Attributes
+    ----------
+    log
+    reader
+    writer
+    reply_terminator
+    command_terminator
+    lock
+    host
+    port
+    timeout
+    connected
+    """
+
+    def __init__(self) -> None:
+        self.log = logging.getLogger(__name__)
+        self.reader = None
+        self.writer = None
+        self.reply_terminator = b"\n"
+        self.command_terminator = "\r"
+        self.lock = asyncio.Lock()
+        self.host = tcpip.LOCAL_HOST
+        self.port = 9999
+        self.timeout = 2
+        self.connected = False
+
+    async def connect(self):
+        """Connect to the electrometer"""
+        self.reader, self.writer = await asyncio.open_connection(
+            host=self.host, port=int(self.port)
+        )
+        self.connected = True
+
+    async def disconnect(self):
+        """Disconnect from the electrometer."""
+        self.reader = None
+        if self.writer is not None:
+            await tcpip.close_stream_writer(self.writer)
+        self.connected = False
+
+    async def send_command(self, msg, has_reply=False):
+        """Send a command to the electrometer and read reply if has one.
+
+        Parameters
+        ----------
+        msg
+        has_reply
+
+        Returns
+        -------
+        reply
+        """
+        msg = msg + self.command_terminator
+        msg = msg.encode("ascii")
```

Comment:
Maybe `msg = msg.encode("ascii") + self.command_terminator` if you set `self.command_terminator = b"\r"`. This might make the code more consistent because you have the `self.reply_terminator = b"\n"` already.